src/web/rustdoc.rs

```diff
@@ -206,12 +208,23 @@ impl RustdocPage {
             .expect("missing TemplateData from the request extensions");
 
         // Build the page of documentation
-        let ctx = ctry!(req, tera::Context::from_serialize(self),);
+        let ctx = ctry!(req, tera::Context::from_serialize(self));
         // Extract the head and body of the rustdoc file so that we can insert it into our own html
-        let html = ctry!(
-            req,
-            utils::rewrite_lol(rustdoc_html, max_parse_memory, ctx, templates)
-        );
+        // while logging OOM errors from html rewriting
+        let html = match utils::rewrite_lol(rustdoc_html, max_parse_memory, ctx, templates) {
+            Err(RewritingError::MemoryLimitExceeded(..)) => {
+                crate::web::metrics::HTML_REWRITE_OOMS.inc();
+
+                let config = extension!(req, Config);
+                let err = failure::err_msg(format!(
+                    "Failed to serve the rustdoc file '{}' because rewriting it surpassed the memory limit of {} bytes",
+                    file_path, config.max_parse_memory,
+                ));
+
+                ctry!(req, Err(err))
```

Comment:
This is basically the same as the `MemoryLimitExceededErrror` [Display impl](https://docs.rs/lol_html/0.2.0/src/lol_html/memory/limiter.rs.html#11), can you use `.context()` instead? That also keeps the `failure` backtrace.

```suggestion
                let err = err.context(format!(
                    "failed to serve the rustdoc file '{}' with memory limit of {} bytes",
                    file_path, max_parse_memory,
                ));

                ctry!(req, Err(err))
```

Reply:
I originally tried this but got the error of
```rust
error[E0599]: no method named `context` found for enum `std::result::Result<std::vec::Vec<u8>, lol_html::rewriter::RewritingError>` in the current scope
   --> src\web\rustdoc.rs:219:31
    |
219 |                 let err = err.context(format!(
    |                               ^^^^^^^ method not found in `std::result::Result<std::vec::Vec<u8>, lol_html::rewriter::RewritingError>`
    |
   ::: C:\Users\Chase\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\src\libcore\result.rs:247:1
    |
247 | pub enum Result<T, E> {
    | --------------------- doesn't satisfy `_: failure::result_ext::ResultExt<std::vec::Vec<u8>, lol_html::rewriter::RewritingError>`
    |
   ::: C:\Users\Chase\.cargo\registry\src\github.com-1ecc6299db9ec823\lol_html-0.2.0\src\rewriter\mod.rs:60:1
    |
60  | pub enum RewritingError {
    | ----------------------- doesn't satisfy `lol_html::rewriter::RewritingError: failure::Fail`
    |
   ::: C:\Users\Chase\.cargo\registry\src\github.com-1ecc6299db9ec823\quick-error-1.2.3\src\lib.rs:983:8
    |
983 |     fn context<X>(self, x: X) -> Result<T, Context<X, E>>;
    |        -------
    |        |
    |        the method is available for `std::boxed::Box<std::result::Result<std::vec::Vec<u8>, lol_html::rewriter::RewritingError>>` here
    |        the method is available for `std::sync::Arc<std::result::Result<std::vec::Vec<u8>, lol_html::rewriter::RewritingError>>` here
    |        the method is available for `std::rc::Rc<std::result::Result<std::vec::Vec<u8>, lol_html::rewriter::RewritingError>>` here
    |
    = note: the method `context` exists but the following trait bounds were not satisfied:
            `lol_html::rewriter::RewritingError: failure::Fail`
            which is required by `std::result::Result<std::vec::Vec<u8>, lol_html::rewriter::RewritingError>: failure::result_ext::ResultExt<std::vec::Vec<u8>, lol_html::rewriter::RewritingError>`
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use quick_error::ResultExt;`

warning: unused import: `failure::ResultExt`
   --> src\web\rustdoc.rs:218:21
    |
218 |                 use failure::ResultExt;
    |                     ^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unused_imports)]` on by default
```

Reply:
Ah right, this is https://github.com/cloudflare/lol-html/issues/54. Don't worry about it then, we can add it later.